### PR TITLE
Feature: Initial file structures for prolog project

### LINKED ISSUE #51: Fazer configuração inicial do projeto

### LINKED ISSUE #54: Obter tamanhos do terminal

### PR DESCRIPTION
**Pull request by augustonz**
### Features
* Added initial files mimicking Haskell project

closes #51

### CODE REVIEW

File: Prolog/models/gameController.pl [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+    %Save high score
+
+    NextTime is Time + 1,
+    sleep(1),
+    Message = "Pressed Enter this amount: " + StateWithInput,
+    write(Message),nl,
+    ttyflush,
+    run(StateWithInput,NextTime).
+
+/*{-# LANGUAGE BlockArguments #-}
```
Comment (diego-aquino) [inline Prolog/models/gameController.pl]: This Haskell code is necessary in this file?

File: Prolog/models/terminal.pl [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+handlePlayerInput(Input):- thread_send_message(main,Input).
+
+fetchFromThread(Resp):- thread_get_message(main,Resp,[timeout(0)]).
+fetchFromThread(Resp):- thread_get_message(main,Resp,[timeout(0)]); Resp = "Nothing".
+
+/*
+
+
+interruptSignal :: Char
```
Comment (diego-aquino) [inline Prolog/models/terminal.pl]: This Haskell code is necessary in this file?

File: Prolog/models/terminal.pl [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  getPlayerInput.
+
+get_key(X):-                                         
+  ttyflush, 
+  get_single_char(X).
+
+handlePlayerInput(Input):- thread_send_message(main,Input).
+
+fetchFromThread(Resp):- thread_get_message(main,Resp,[timeout(0)]).
+fetchFromThread(Resp):- thread_get_message(main,Resp,[timeout(0)]); Resp = "Nothing".
```
Comment (diego-aquino) [inline Prolog/models/terminal.pl]: Nice!

File: Prolog/models/terminal.pl [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+startPlayerInputThread:- 
+thread_create(getPlayerInput, Id).
+
+getPlayerInput:- 
+  get_key(NewInput),
+  handlePlayerInput(NewInput),
+  getPlayerInput.
+
+get_key(X):-
```
Comment (diego-aquino) [inline Prolog/models/terminal.pl]: Are we going to follow `camelCase`? If so, what do you think about renaming this rule to `getKey` for consistency?

File: Prolog/models/gameController.pl (hunk 1 above)
Comment (augustonz) [reply Prolog/models/gameController.pl]: I was using the code as reference and forgot to take it out, my baad

File: Prolog/models/terminal.pl (hunk 4 above)
Comment (augustonz) [reply Prolog/models/terminal.pl]: sure thing!

**Review by Murilo-Gruppi — APPROVED** [review]

**Review by diego-aquino — APPROVED** [review]
Good job!

**Pull request merged**